pkg/sensors/tracing/genericuprobe.go

```diff
@@ -479,50 +566,115 @@ func addUprobe(spec *v1alpha1.UProbeSpec, ids []idtable.EntryID, in *addUprobeIn
+		if spec.ReturnArg == nil {
+			return nil, errors.New("ReturnArg not specified with Return=true")
+		}
+		argType := gt.GenericTypeFromString(spec.ReturnArg.Type)
+		if argType == gt.GenericInvalidType {
+			if spec.ReturnArg.Type == "" {
+				return nil, errors.New("ReturnArg not specified with Return=true")
+			}
+			return nil, fmt.Errorf("ReturnArg type '%s' unsupported", spec.ReturnArg.Type)
+		}
+		eventConfig.ArgReturn = int32(argType)
+		argP := argPrinter{index: api.ReturnArgIndex, ty: argType}
+		argReturnPrinters = append(argReturnPrinters, argP)
+	} else {
+		eventConfig.ArgReturn = int32(gt.GenericUnsetType)
+	}
+
+	if argRetprobe != nil {
+		setRetprobe = true
+
+		argType := gt.GenericTypeFromString(argRetprobe.Type)
+		eventConfig.ArgReturnCopy = int32(argType)
+
+		argP := argPrinter{index: int(argRetprobe.Index), ty: argType, label: argRetprobe.Label}
+		argReturnPrinters = append(argReturnPrinters, argP)
+	} else {
+		eventConfig.ArgReturnCopy = int32(gt.GenericUnsetType)
+	}
+
+	// Write attributes into BTF ptr for use with load
+	if !setRetprobe {
+		setRetprobe = spec.Return
+	}
+
+	addUprobeEntry := func(sym string, offset uint64, idx int) error {
 		var refCtrOffset uint64
 
 		if refCtrOffsets != 0 {
 			refCtrOffset = spec.RefCtrOffsets[idx]
 		}
 
-		config := initEventConfig()
-		config.ArgType = argTypes
-		config.ArgMeta = argMeta
-		config.ArgIndex = argIdx
+		eventConfig.ArgType = argTypes
+		eventConfig.ArgMeta = argMeta
+		eventConfig.ArgIndex = argIdx
 
 		uprobeEntry := &genericUprobe{
-			tableId:      idtable.UninitializedEntryID,
-			config:       config,
-			path:         spec.Path,
-			symbol:       sym,
-			address:      offset,
-			refCtrOffset: refCtrOffset,
-			selectors:    uprobeSelectorState,
-			policyName:   in.policyName,
-			message:      msgField,
-			argPrinters:  argPrinters,
-			tags:         tagsField,
+			loadArgs: uprobeLoadArgs{
+				retprobe: setRetprobe,
+				config:   eventConfig,
+				selectors: kprobeSelectors{
+					entry: uprobeSelectorState,
+					retrn: uprobeRetSelectorState,
+				},
+			},
+			tableId:           idtable.UninitializedEntryID,
+			path:              spec.Path,
+			symbol:            sym,
+			address:           offset,
+			refCtrOffset:      refCtrOffset,
+			policyName:        in.policyName,
+			message:           msgField,
+			argPrinters:       argPrinters,
+			argReturnPrinters: argReturnPrinters,
+			tags:              tagsField,
+			pendingEvents:     nil,
+		}
+
+		uprobeEntry.pendingEvents, err = lru.New[pendingEventKey, pendingEvent[*tracing.MsgGenericUprobeUnix]](4096)
+		if err != nil {
+			return err
 		}
 
 		uprobeTable.AddEntry(uprobeEntry)
 		id := uprobeEntry.tableId
 
-		config.FuncId = uint32(id.ID)
+		eventConfig.FuncId = uint32(id.ID)
 
 		ids = append(ids, id)
+		return nil
 	}
 
 	if symbols != 0 {
 		for idx, sym := range spec.Symbols {
-			if err := checkSymbol(sym); err != nil {
+			if err = checkSymbol(sym); err != nil {
```

Comment:
why is this needed? there's other places like that you did not change.. I think this could stay?